### PR TITLE
Add midori databases

### PR DESCRIPTION
**Pull request by d4straub**
Addresses partially https://github.com/nf-core/ampliseq/issues/427 by adding the MIDORI2 CO1 database.

In the course of this, I added another optional field `taxlevels` to `conf/ref_databases.config`, which is similar to `--dada_assign_taxlevels`, and allows defining the taxonomic levels in the standard databases; `--dada_assign_taxlevels` overwrites  `taxlevels`.

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
  - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/ampliseq/tree/master/.github/CONTRIBUTING.md)
  - [ ] If necessary, also make a PR on the nf-core/ampliseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: conf/ref_databases.config [hunk 1]
```diff
@@ -9,6 +9,13 @@
 
 params {
     dada_ref_databases {
+        'midori2-co1=gb250' {
+            title = "MIDORI2 - CO1 Taxonomy Database - Release GB250"
+            file = [ "http://reference-midori.info/download/Databases/GenBank250/DADA2/uniq/MIDORI2_UNIQ_NUC_GB250_CO1_DADA2.fasta.gz", "http://reference-midori.info/download/Databases/GenBank250/DADA2_sp/uniq/MIDORI2_UNIQ_NUC_SP_GB250_CO1_DADA2.fasta.gz" ]
+            citation = "Machida RJ, Leray M, Ho SL, Knowlton N. Metazoan mitochondrial gene sequence reference datasets for taxonomic assignment of environmental samples. Sci Data. 2017 Mar 14;4:170027. doi: 10.1038/sdata.2017.27. PMID: 28291235; PMCID: PMC5349245."
+            fmtscript = "taxref_reformat_midori2.sh"
+            dbversion = 'midori2-co1=gb250'
+        }
```
Comment (erikrikarddaniel) [inline conf/ref_databases.config:18]: We should have a generic without version number, right? We do for the other databases.

**Review by erikrikarddaniel — APPROVED** [review]
Just a small comment.

**Comment by d4straub** [comment]
Thanks, you are totally right!

Unfortunately `MIDORI2_UNIQ_NUC_GB250_CO1_DADA2.fasta.gz` (for assignTaxonomy) starts at class, i.e. lacks "Domain", "Kingdom", "Phylum" specified in https://github.com/nf-core/ampliseq/blob/aed35e5b4bad907ed2bdf6587efb82ad2578e6b1/modules/local/dada2_taxonomy.nf#L30 which can be fixed, but isnt nice. I'd need to copy class 3 times to come to this results, seems not very intuitive.

An example fasta name of `MIDORI2_UNIQ_NUC_GB250_CO1_DADA2.fasta.gz` is:
`>Discosea_555280;Flabellinia_1485085;order_Vannellidae_95227;Vannellidae_95227;Clydonella_218657;Clydonella sawyeri_2201168;`

File: conf/ref_databases.config (hunk 1 above)
Comment (d4straub) [reply conf/ref_databases.config]: done!

**Comment by d4straub** [comment]
Tested and works as far as I can judge, so I'll merge now.

**Pull request merged**